pkg/encryption/common.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 PingCAP, Inc. Licensed under Apache-2.0.
+
+package encryption
+
+import (
+	"crypto/rand"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/spacemonkeygo/openssl"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/backup"
+	"github.com/pingcap/kvproto/pkg/encryptionpb"
+	"github.com/pingcap/log"
+)
+
+const (
+	keyBackendTypeFile = "file"
+
+	metaEncryptedKey = "encrypted_key"
+	metaKeyIV        = "iv"
+	metaKeyTag       = "tag"
+
+	keySize        = 32
+	ivSize         = 12
+	dataFileIVSize = 16
+)
+
+// Options contain common encryption configurations.
+type Options struct {
+	fileBackendOptions
+
+	Config         encryptionpb.EncryptionConfig
+	keyBackendType string
+}
+
+type keyBackend interface {
+	GetKey() (key []byte, encryptedKey []byte, err error)
+	DecryptKey(encryptedKey []byte) (key []byte, err error)
+}
+
+func (options *Options) fillDataKey() error {
```

Comment:
According to discussion on the docs, will change this to derive the key from backupmeta key using HKDF: https://docs.google.com/document/d/1x5xx8Zbm-frzj1-t5OnMcpEUQHjpnl6vSkCmaehpeLM/edit?disco=AAAAJZ6wItk

Reply:
(note that the Google doc is not public. we should move these design documents into this repo in the future.)